pages/en/lb4/Application.md

```diff
@@ -0,0 +1,211 @@
+tags:
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/Application.html
+summary:
+---
+
+## What is an Application?
+
+In LoopBack 4, the [`Application`](http://apidocs.strongloop.com/@loopback%2fcore/#Application)
+class is the central class for setting up all of your module's components,
+controllers, servers and bindings. The `Application` class extends 
+[Context](Context.html), and provides the controls for starting and stopping
+itself and its associated servers.
+
+When using LoopBack 4, we strongly encourage you to create your own subclass
+of `Application` to better organize your configuration and setup.
+
+## Making your own application class
+
+By making your own application class, you can perform several additional
+tasks as a part of your setup:
+- Pass configuration into the base class constructor
+- Perform some asynchronous wireup before application start
+- Perform some graceful cleanup on application stop 
+
+{% include code-caption.html content="src/widget-application.ts" %}
+```ts
+import {Application} from '@loopback/core';
+import {RestComponent, RestServer} from '@loopback/rest';
+import {SamoflangeController, DoohickeyController} from './src/controllers';
+import {WidgetApi} from './apidef/';
+
+export class WidgetApplication extends Application {
+  constructor() {
+    // This is where you would pass configuration to the base constructor
+    // (as well as handle your own!)
+    super({
+      components: [RestComponent],
+    });
+    const app = this; // For clarity.
+    // You can bind to the Application-level context here.
+    // app.bind('foo').to(bar);
+    app.controller(SamoflangeController);
+    app.controller(DoohickeyController);
+  }
+
+  async start() {
+    // This is where you would asynchronously retrieve servers, providers and
+    // other components to configure them before launch.
+    const server = await app.getServer(RestServer);
+    server.bind('rest.port').to(8080);
+    server.api(WidgetApi);
+    return await super.start();
+  }
+
+  async stop() {
+    // This is where you would do whatever is necessary before stopping your
+    // app (graceful closing of connections, flushing buffers, etc)
+    console.log('Widget application is shutting down...')
+    await super.stop();
+  }
+}
+
+```
+
+## Configuring your application
+Your application can be configured with constructor arguments, bindings, or
+a combination of both.
+
+### Binding configuration
+Binding is the most commonly-demonstrated form of application configuration
+throughout our examples, and in most cases, is the recommended method for
+setting up your application.
+
+In addition to the binding functions provided by [Context](Context.html),
+the `Application` class also provides some sugar functions for commonly used
+bindings, like `component`, `server` and `controller`:
+
+```ts
+export class MyApplication extends Application {
+  constructor() {
+    super();
+  }
+  this.component(MagicSuite);
```

Comment:
This is outside of `constructor` function, inside class declaration - looks like a typo to me? Please check that the snippet compiles.